### PR TITLE
Smarti Version 0.1.0

### PR DESCRIPTION
**Pull request by tkurz**
[NEW]

Hi @mrsimpson 

Hi, here comes the first version of the Smarti Plugin. Basically it includes a new Knowledge Base Adapter for the Smarti Backend. Furthermore it includes a lightweight client part which loads a script provided by Smarti into the Rocket.Chat frontend. The actual Frontend Script uses the Realtime API (slightly extended by this Plugin). The Plugin can be configured via the Rockte.Chat admin UI. I am not sure if I included the "close room" operation correctly, so please took a closer look into that. In add

### CODE REVIEW

File: packages/assistify/config.js [hunk 1]
```diff
@@ -50,6 +51,14 @@ Meteor.startup(()=>{
 		i18nLabel: 'DBS_AI_Redlink_Auth_Token'
 	});
 
+	RocketChat.settings.add('DBS_AI_Redlink_Hook_Token', '', {
```
Comment (mrsimpson) [inline packages/assistify/config.js:54]: @tkurz what is the *configured* token used for?

File: packages/assistify/package.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ Package.onUse(function(api) {
 
 	//Server
 	api.addFiles('config.js', 'server');
+	//TODO add jquery?
```
Comment (mrsimpson) [inline packages/assistify/package.js:20]: jQuery is part of Blaze (the templating engine) (@see .meteor/local/build/programs/web.browser/packages/blaze.js)
still, requiring it explicitly should do no harm `api.use('jquery', 'client');`

File: packages/assistify-help-request/client/views/tabbar/HelpRequestActions.js [hunk 3]
```diff
@@ -115,7 +115,8 @@ Template.HelpRequestActions.events({
 					});
 
 					instance.helpRequest.set(
-						RocketChat.models.HelpRequests.findOneByRoomId(Template.currentData())
+						//RocketChat.models.HelpRequests.findOneByRoomId(Template.currentData())
+						RocketChat.models.HelpRequests.findOneByRoomId(instance.data.roomId)//TODO is this correct?
```
Comment (mrsimpson) [inline packages/assistify-help-request/client/views/tabbar/HelpRequestActions.js:119]: @tkurz it should be

File: packages/dbs-ai/client/views/app/tabbar/smarti.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+			var WEBSOCKET_URL =
+				"ws" + RocketChat.settings.get('Site_Url').substring(4) + "websocket/";
+
+
+			self.smarti = new window.SmartiWidget(self.find('.external-message'), {
+				socketEndpoint: WEBSOCKET_URL,
+				smartiEndpoint: DBS_AI_Redlink_URL,
+				channel: self.data.rid,
+				inputCssSelector:'.autogrow-shadow'
```
Comment (mrsimpson) [inline packages/dbs-ai/client/views/app/tabbar/smarti.js:53]: @tkurz This is about the input box (in order to copy to), right? if so, what about `$('#chat-window-${ instance.data.roomId } .input-message')`

File: packages/dbs-ai/server/config.js [hunk 5]
```diff
@@ -39,6 +40,13 @@ Meteor.startup(function() {
 		i18nLabel: 'DBS_AI_Redlink_Auth_Token'
 	});
 
+	RocketChat.settings.add('DBS_AI_Redlink_Hook_Token', 'key123', {
```
Comment (mrsimpson) [inline packages/dbs-ai/server/config.js:43]: @tkurz difference to the setting in https://github.com/mrsimpson/Rocket.Chat/pull/34/files#diff-8e83d6a3163a81272f59768476959fa7 ?

File: packages/dbs-ai/server/lib/Smarti.js [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+			} else {
+				SystemLogger.error('Smarti - analysis triggering failed:', JSON.stringify(response, '', 2));
+			}
+		} catch (e) {
+			SystemLogger.error('Smarti response did not succeed: ', e);
+		}
+	}
+
+	onClose(room) { //async
+		//TODO add options here?
```
Comment (mrsimpson) [inline packages/dbs-ai/server/lib/Smarti.js:54]: @tkurz which options do you mean? Header options?
What about transmitting a context?

File: packages/dbs-ai/server/lib/Smarti.js [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+RocketChat.API.v1.addRoute('smarti.result/:token', {authRequired: false}, {
+	post() {
+
+		check(this.bodyParams, Match.ObjectIncluding({
+			conversationId: String,
+			channelId: String
+		}));
+
+		//verify token
+		if(this.urlParams.token && this.urlParams.token == RocketChat.settings.get('DBS_AI_Redlink_Hook_Token')) {
```
Comment (mrsimpson) [inline packages/dbs-ai/server/lib/Smarti.js:161]: ah, is the hook token a "secret"? Then, we should at least md5-it, not?

**Review by mrsimpson — CHANGES_REQUESTED** [review]
Hi @tkurz ,
please find some comments included. Most are minor. Submitting a context to Smarti should be possible though. Also I wonder about the redundant setting.
Kindly revise.
Cheers, Oliver

**Comment by mrsimpson** [comment]
@tkurz and can you please PR to mrsimpson:develop?

File: packages/assistify/config.js (hunk 1 above)
Comment (tkurz) [reply packages/assistify/config.js:54]: it is used to verify if a analysis result is send by a trusted source

File: packages/assistify/package.js (hunk 2 above)
Comment (tkurz) [reply packages/assistify/package.js:20]: the question was about jquery which is used by the bundled smarti script. It is included/packaged there which is maybe also not a good solution. requirejs would be the best solution?

Comment (mrsimpson) [reply packages/assistify/package.js:20]: @tkurz: already undestood. In my reply, you can find the declaration of dependency which will make sure jquery is available.

Comment (mrsimpson) [reply packages/assistify/package.js:20]: @tkurz: already undestood. In my reply, you can find the declaration of dependency which will make sure jquery is available.

**Comment by mrsimpson** [comment]
merged to develop, test it in an integrated manner

**Pull request closed without merging**